### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by IjzerenHein**
gif

## Task

Add basic tests to validate the 4 configurations. The tests verify that the eslint & typescript config can be loaded and processed correctly. This PR will help us to verify upgrades to newer dependencies.

## Other changes

- I upgraded the github actions to `@4`

## Test plan

- Added `yarn test`
- Runs `yarn test` in CI on every PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +27,8 @@
     ".prettierrc.js"
   ],
   "devDependencies": {
-    "eslint": "8.49.0"
+    "eslint": "8.49.0",
+    "@types/node": "20"
```
Comment (isilher) [inline package.json:31]: in `.nvmrc` we specify this package to be on node 18. Can you either update that one too or pin this to `"18"`?

**Review by isilher — COMMENTED** [review]
Awesome! This will help us a lot 🦾

Just one note on node 😄

File: package.json (hunk 1 above)
Comment (IjzerenHein) [reply package.json]: Ah good catch, Ive updated it 👍 👍

**Review by isilher — APPROVED** [review]
💯

**Pull request merged**